v2/contribute/decisions/thirdparty/0003-provider-specific-config-validation-in-core.mdx

```diff
@@ -0,0 +1,46 @@
+---
+id: "0003"
+title: Validation of provider specific config in the core
+hide_title: true
+---
+
+import DecisionHeader from "/src/components/decisionLogs/DecisionHeader"
+import ArgumentPro from "/src/components/decisionLogs/ArgumentPro"
+import ArgumentCon from "/src/components/decisionLogs/ArgumentCon"
+import ArgumentNeut from "/src/components/decisionLogs/ArgumentNeut"
+
+# Validation of AdditionalConfig of a provider config in the core
+
+<DecisionHeader status="proposed" lastUpdate="2022-12-08" created="2022-12-08" deciders={["rishabhpoddar", "sattvikc"]} proposedBy={["sattvikc"]} />
+
+## Context and Problem Statement
+
+Should the additionalConfig (provider specific) (such as `hd` for google workspaces, `keyId`, `teamId` etc for apple) validated in core, while adding or updating them?
+
+## Considered Options
+
+* Yes, Validate in core
+* No provider specific validation in core
+
+
+## Decision Outcome
+
+Chosen option: **No provider specific validation in core**, because
```

Comment:
We decided to not do this and pick validation in core cause at least for the providers that the core knows about, we can ensure that the user has the correct configs. For new / unknown providers (to the core), they will not be checked anyway (which would then be equivalent to option 2). So there is no bug disadvantage to validating config specific providers. 

The only one being that we get the config specific checks incorrect / incomplete in the core. This also we can solve by providing a flag to the API which will force no checks.

Also note that the core should allow config input with any empty clientType array since the user could have added those clientType config on the backend SDK level.

Reply:
Done